editoast/src/views/layers/layer.rs

```diff
@@ -91,6 +95,65 @@ pub async fn layer_view(
     }))
 }
 
+/// Gets mvt tile from the cache if possible, otherwise gets data fro the data base and caches it in redis
+#[get(
+    "/tile/<layer_slug>/<view_slug>/<z>/<x>/<y>?<infra>",
+    format = "application/octet-stream"
+)]
+#[allow(clippy::too_many_arguments)]
+pub async fn cache_and_get_mvt_tile<'a>(
+    layer_slug: &str,
+    view_slug: &str,
+    z: u64,
+    x: u64,
+    y: u64,
+    infra: i64,
+    map_layers: &State<MapLayers>,
+    conn: DBConnection,
+    pool: &RedisPool,
+) -> ApiResult<Vec<u8>> {
+    let layer = match map_layers.layers.get(layer_slug) {
+        Some(layer) => layer,
+        None => return Err(LayersError::new_layer_not_found(layer_slug, map_layers).into()),
+    };
+    let view = match layer.views.get(view_slug) {
+        Some(view) => view,
+        None => return Err(LayersError::new_view_not_found(view_slug, layer).into()),
+    };
+    let cache_key = get_cache_tile_key(
+        &get_view_cache_prefix(layer_slug, infra, view_slug),
+        &Tile { x, y, z },
+    );
+    let cached_value = get::<Vec<u8>>(pool, &cache_key).await;
+
+    if let Some(value) = cached_value {
+        return Ok(value);
+    }
+
+    let geo_json_query = get_geo_json_sql_query(&layer.table_name, view);
+    let records = conn
+        .run::<_, ApiResult<_>>(move |conn| {
+            match sql_query(geo_json_query)
+                .bind::<Integer, _>(z as i32)
+                .bind::<Integer, _>(x as i32)
+                .bind::<Integer, _>(y as i32)
+                .bind::<Integer, _>(infra as i32)
+                .get_results::<GeoJsonAndData>(conn)
+            {
+                Ok(results) => Ok(results),
+                Err(err) => Err(err.into()),
+            }
+        })
+        .await?;
+    let mvt_bytes: Vec<u8> = create_and_fill_mvt_tile(z, x, y, layer_slug, records)
+        .to_bytes()
+        .unwrap();
+    set(pool, &cache_key, mvt_bytes.clone())
+        .await
+        .unwrap_or_else(|_| panic!("Fail to set value in redis with key {cache_key}"));
```

Comment:
```suggestion
        .expect(format!("Fail to set value in redis with key {cache_key}"));
```

Reply:
this is catch by clippy I think suggesting to use `unwrap_or_else` instead